framework/src/userobject/ShapeElementUserObject.C

```diff
@@ -0,0 +1,58 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ShapeElementUserObject.h"
+#include "Assembly.h"
+
+template<>
+InputParameters validParams<ShapeElementUserObject>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  params.addParam<bool>("compute_jacobians", true, "Compute Jacobians for coupled variables");
+  params.addParamNamesToGroup("compute_jacobians", "Advanced");
+  return params;
+}
+
+ShapeElementUserObject::ShapeElementUserObject(const InputParameters & parameters) :
+    ElementUserObject(parameters),
+    _phi(_assembly.phi()),
+    _grad_phi(_assembly.gradPhi()),
+    _currently_computing_jacobian(_fe_problem.currentlyComputingJacobian()),
```

Comment:
This is invalid.  `FEProblem::currentlyComputingJacobian()` returns a `bool`... NOT a `bool &`... so you're setting a reference to a temporary.

That said: I don't mind this idea (of having `currentlyComputingJacobian()` return a `bool &` to be used in this way)... that's just not how it works right now.